finances.js

```diff
@@ -14,24 +14,92 @@ let bankAccount = {
 
 function main() {
   console.log("Welcome to your personal finance manager");
+
   while (true) {
     console.log(
       "What do you want to do?\n" +
-      "[b] show latest balance\n" +
-      "[bd]show balance for a specific date\n" +
-      "[t] show all transactions\n" +
-      "[a] add a transaction\n" +
-      "[d] delete a transaction\n" +
-      "[s] show statistics\n" +
-      "[q] quit"
+        "[b] show latest balance\n" +
+        "[bd]show balance for a specific date\n" +
+        "[t] show all transactions\n" +
+        "[a] add a transaction\n" +
+        "[d] delete a transaction\n" +
+        "[s] show statistics\n" +
+        "[q] quit"
     );
     let choice = prompt();
     if (choice === "q") {
       break;
     } else if (choice === "b") {
       showCurrentBalance();
+    } else if (choice === "bd") {
+      showBalanceForSpecificDate();
+    } else if (choice === "t") {
+      showAllTransactions();
+    } else if (choice === "a") {
+      addATransaction();
+    } else if (choice === "d") {
+      deleteATransaction();
+    }
+  }
+}
+
+function showCurrentBalance() {
+  let sum = bankAccount.startBalance;
+  for (let allTransactions of bankAccount.transactions) {
+    sum += allTransactions.amount;
+  }
+  console.log(`Current balance: ${sum}`);
+}
+
+function showBalanceForSpecificDate() {
+  console.log("Please insert the date (e.g 2022-09-31) ");
+  let balanceOfSpecificDate = 0;
+  let userInputDate = prompt();
+  for (let allTransactions of bankAccount.transactions) {
+    if (allTransactions.date === userInputDate) {
+      balanceOfSpecificDate += allTransactions.amount;
     }
   }
+  console.log(balanceOfSpecificDate);
+}
+
+function showAllTransactions() {
+  for (let showTransactions of bankAccount.transactions) {
+    console.log(showTransactions);
+  }
+}
+
+function addATransaction() {
+  let dateOfNewTransaction = prompt("Please insert the date, on which the transaction was made.(year-month-day) ");
+  let amountOfNewTransaction = prompt("Please insert how much money you spent or have received. (+/- number) " );
+  let titleOfNewTransaction = prompt("Please insert on what you spent your money on ");
+  let newTransaction = {
+    id: bankAccount.transactions.length + 1,
+    date: dateOfNewTransaction,
+    amount: parseInt(amountOfNewTransaction),
+    title: titleOfNewTransaction
+  };
+  bankAccount.transactions.push(newTransaction);
+}
+
+function deleteATransaction() {
+  console.log("Delete a transaction by writing a number starting with 1 for the following transaction:  ");
+  for (let i = 0; i < bankAccount.transactions.length; i++) {
+    console.log("Press " + (i + 1) + " to delete  ");
+    console.log(bankAccount.transactions[i]);
+  }
+  let userInput = prompt();
+  parseInt(userInput);
+  bankAccount.transactions.splice(userInput - 1, 1);
+  for (let j = 0; j < bankAccount.transactions.length; j++) {
+    bankAccount.transactions[j].id = bankAccount.transactions[j].id - 1;
+  }
```

Comment:
Achtung: hier wird die ID von _jeder_ Transaktion verringert.

Du willst damit wohl eine Lücke in den IDs vermeiden, auf diese Art verschwindet die Lücke aber nicht.

Wenn ich aber über die Angabe nachdenke, denk ich aber jetzt, wir brauchen eigentlich gar keine IDs. D.h. du könntest das ID-Feld auch einfach ignorieren bzw raus löschen und einfach immer den Array-Index verwenden. Das ist im Nachhinein etwas, was vielleicht gar nicht in die Angabe schreiben hätte sollen.

Reply:
Ich denke,ich habe es jetzt gelöst. (hoffe ich)
Indem ich bei der function showAllTransactions  noch  bankAccount.transactions[i].id = i+1; definiere in der for-Schleife.
Ich lasse sonst die IDs noch drinnen.